File: binary-search/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  let startIndex = 0;
  let endIndex = nums.length - 1;
  while (startIndex <= end) {
    const midIndex = Math.floor((startIndex + endIndex) / 2);
    if (nums[midIndex] === target) {
      return mid;
    }
    if (nums[midIndex] > target) {
      endIndex = midIndex - 1;
    } else {
      startIndex = midIndex + 1;
    }
  }
  return -1;
};

File: easy/binary-search/binary-search.c
int search(int *nums, int numsSize, int target)
{
    int startIndex = 0;
    int endIndex = numsSize - 1;
    while (startIndex <= endIndex)
    {
        int midIndex = (startIndex + endIndex) / 2;
        if (nums[midIndex] == target)
        {
            return midIndex;
        }
        if (nums[midIndex] > target)
        {
            endIndex = midIndex - 1;
        }
        else
        {
            startIndex = midIndex + 1;
        }
    }
    return -1;
}